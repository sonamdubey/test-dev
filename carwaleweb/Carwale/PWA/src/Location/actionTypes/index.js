export const OPEN_POPUP = 'OPEN_POPUP';
export const CLOSE_POPUP = 'CLOSE_POPUP';
export const SHOW_CROSS_ICON = 'SHOW_CROSS_ICON';
export const HIDE_CROSS_ICON = 'HIDE_CROSS_ICON';
