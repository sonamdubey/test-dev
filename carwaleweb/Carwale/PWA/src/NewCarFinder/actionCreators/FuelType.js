import 'isomorphic-fetch'

import {
	REQUEST_FUEL_TYPE,
	RECEIVE_FUEL_TYPE,
	FUEL_TYPE_SELECTION,
	SET_FUEL_STORED_FILTERS
} from '../actionTypes'

import FuelTypeApi from '../apis/FuelType'

import {
	getCityId
} from '../../selectors/LocationSelectors'

import {
	getBudgetFilter,
	getBodyTypeFilter
} from '../selectors/NCFSelectors'

export const requestFuelType = () => {
	return {
		type: REQUEST_FUEL_TYPE
	}
}

export const setFuelTypeData = data => {
	return {
		type: RECEIVE_FUEL_TYPE,
		data
	}
}

export const selectFuelType = id => {
	return {
		type: FUEL_TYPE_SELECTION,
		id
	}
}

export const fetchFuelType = isFetching => (dispatch, getState) => {
	dispatch(requestFuelType(isFetching))
	const state = getState()
	const options = {...getCityId(state), ...getBudgetFilter(state), ...getBodyTypeFilter(state)}
	FuelTypeApi
		.get(options)
		.then( data => { setTimeout(() => dispatch(setFuelTypeData(data)), 300)})
		.catch( error => {
			//TODO: how should error affect the state ?
			console.log(error)
		})
}


export const setFuelStoredFilters = storedFilters => {
	return {
		type: SET_FUEL_STORED_FILTERS,
		storedFilters
	}
}
