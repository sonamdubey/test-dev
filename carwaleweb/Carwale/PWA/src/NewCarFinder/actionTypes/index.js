export const NEXT_FILTER = 'NEXT_FILTER'
export const SET_CURRENT_SCREEN = 'SET_CURRENT_SCREEN'
export const SET_ALL_FILTER_SCREEN = 'SET_ALL_FILTER_SCREEN'

export const REQUEST_CAR_FEATURE = 'REQUEST_CAR_FEATURE'
export const RECEIVE_CAR_FEATURE = 'RECEIVE_CAR_FEATURE'
export const CAR_FEATURE_SELECTION = 'CAR_FEATURE_SELECTION'

// budget
export const UPDATE_BUDGET_SLIDER_VALUE = 'UPDATE_BUDGET_SLIDER_VALUE'
export const UPDATE_BUDGET_INPUT_VALUE = 'UPDATE_BUDGET_INPUT_VALUE'

// body type
export const REQUEST_BODY_TYPE = 'REQUEST_BODY_TYPE'
export const RECEIVE_BODY_TYPE = 'RECEIVE_BODY_TYPE'
export const BODY_TYPE_SELECTION = 'BODY_TYPE_SELECTION'
export const SET_BODY_STORED_FILTERS = 'SET_BODY_STORED_FILTERS'


// fuel type
export const REQUEST_FUEL_TYPE = 'REQUEST_FUEL_TYPE'
export const RECEIVE_FUEL_TYPE = 'RECEIVE_FUEL_TYPE'
export const FUEL_TYPE_SELECTION = 'FUEL_TYPE_SELECTION'
export const SET_FUEL_STORED_FILTERS = 'SET_FUEL_STORED_FILTERS'

// fuel type
export const REQUEST_MAKE = 'REQUEST_MAKE'
export const RECEIVE_MAKE = 'RECEIVE_MAKE'
export const MAKE_SELECTION = 'MAKE_SELECTION'
export const SET_MAKE_STORED_FILTERS = 'SET_MAKE_STORED_FILTERS'



// listing
// model list RECEIVE_MODEL_LIST
export const REQUEST_MODEL_LIST = 'REQUEST_MODEL_LIST'
export const RECEIVE_MODEL_LIST = 'RECEIVE_MODEL_LIST'
export const APPEND_MODEL_LIST = 'APPEND_MODEL_LIST'

// filter screen
export const OPEN_FILTER_SCREEN = 'OPEN_FILTER_SCREEN'
export const CLOSE_FILTER_SCREEN = 'CLOSE_FILTER_SCREEN'

// compare cars
export const ADD_MODEL_TO_COMPARE = 'ADD_MODEL_TO_COMPARE'
export const REMOVE_MODEL_FROM_COMPARE = 'REMOVE_MODEL_FROM_COMPARE'
// header
export const MAKE_HEADER_STICKY = 'MAKE_HEADER_STICKY'
export const MAKE_HEADER_NORMAL = 'MAKE_HEADER_NORMAL'
