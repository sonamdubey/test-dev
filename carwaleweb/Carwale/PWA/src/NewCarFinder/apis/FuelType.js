
import { FUELTYPE_ENDPOINT } from '../constants'
import { serialzeObjectToQueryString } from '../../utils/Common'
/**
 * This returns a promise to fetch
 * fuel types availability for the selected filters
 * @return {Promise<>}
 */
const get = (options) => {
    return fetch(FUELTYPE_ENDPOINT+"?"+serialzeObjectToQueryString(options))
    .then(response => {
        if (!response.ok) { throw response }
        return response.json()
    })
}
const FuelTypeApi = {
    get
}
export default FuelTypeApi
