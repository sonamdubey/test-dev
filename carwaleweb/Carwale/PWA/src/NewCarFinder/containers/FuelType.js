import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import SpeedometerLoader from '../../components/SpeedometerLoader'

import {
	fetchFuelType,
	selectFuelType,
	setFuelStoredFilters
} from '../actionCreators/FuelType'

import {
	setCurrentScreen
} from '../actionCreators/Filter'

import {
	initToast,
	clearToast
} from '../../actionCreators/Toast'

import {
	makeCancelable
} from '../../utils/CancelablePromise'

import {
	getScreenId
} from '../selectors/NCFSelectors'

import ListItem from '../../components/ListItem'

import { fireInteractiveTracking } from '../../utils/Analytics';
import {CATEGORY_NAME} from '../constants/index';
import { trackCustomData } from '../../utils/cwTrackingPwa';
class FuelType extends React.Component {
	constructor(props) {
		super(props)
	}

	componentDidMount() {
		const {
			fetchFuelType,
			setCurrentScreen,
			screenId
		} = this.props

		setCurrentScreen(screenId)
		fetchFuelType()
		trackCustomData(CATEGORY_NAME, "FuelTypeScreenImpression", "NA", false)
	}

	componentWillUnmount() {
		this.props.clearToast()
	}

	componentDidUpdate(prevProps) {
		if(prevProps.isFetching && !this.props.isFetching)
		{
			this.props.setFuelStoredFilters(this.props.storedFueltype)
		}
	}

	getFuelTypeList = (data) => {
		let list = data.map(item => {
			const itemStatusClass = this.getItemStatus(item)

			return (
				<li
					key={item.id}
					className={"fuel-type-list__item " + itemStatusClass}
					onClick={this.handleFuelTypeClick.bind(this, item)}
				>
					<ListItem
						item={item}
					/>
				</li>
			)
		})

		return list
	}

	getItemStatus = (item) => {
		let status = ''

		if (item.isSelected) {
			status = 'active'
		}
		else if (!item.carCount) {
			status = 'disable'
		}

		return status
	}

	handleFuelTypeClick = (item, event) => {
		if (item.carCount) {
			this.props.selectFuelType(item.id)
		}
		else {
			this.setToast(item, event)
		}
		const status = (item.carCount > 0 ? (item.isSelected ? "Deselect" : "Select") : "NotAvailableClick")
		fireInteractiveTracking(CATEGORY_NAME, "NCF_FuelTypeFilter", status+ "_" + item.name)
		trackCustomData(CATEGORY_NAME,"FuelTypeFilterClick","state="+status+"|fuelType="+item.name,false)
	}

	setToast = (item, event) => {
		this.props.initToast({
			message: `There is no ${item.name} car matching your criteria.`,
			event: event
		})
	}

	render() {
		const {
			fuelTypeData,
			isFetching
		} = this.props

		if(isFetching){
			return <SpeedometerLoader />
		}
		let list

		const isDataAvailable = fuelTypeData && fuelTypeData.length

		if (isDataAvailable) {
			list = this.getFuelTypeList(fuelTypeData)
		}

		return (
			<div className="fuel-type-screen">
				<div className="screen__head">
					<p className="screen-head__title">Select your preferred fuel type</p>
				</div>
				<div className="screen__body">
					<ul className="fuel-type__list">
						{list}
					</ul>
				</div>
			</div>
		)
	}
}

const mapStateToProps = (state) => {
	const {
		data: fuelTypeData,
		isFetching
	} = state.newCarFinder.fuelType
	const {
		fuelType: storedFueltype
	} = state.filtersScreen.filters
	const screenId = getScreenId(state,'FuelTypeFilter')
	return {
		fuelTypeData,
		isFetching,
		screenId,
		storedFueltype
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		fetchFuelType: bindActionCreators(fetchFuelType, dispatch),
		selectFuelType: bindActionCreators(selectFuelType, dispatch),
		initToast: bindActionCreators(initToast, dispatch),
		clearToast: bindActionCreators(clearToast, dispatch),
		setCurrentScreen: bindActionCreators(setCurrentScreen, dispatch),
		setFuelStoredFilters: bindActionCreators(setFuelStoredFilters, dispatch)
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(FuelType)
