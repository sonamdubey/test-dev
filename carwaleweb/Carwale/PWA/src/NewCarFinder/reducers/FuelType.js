import {
	REQUEST_FUEL_TYPE,
	RECEIVE_FUEL_TYPE,
	FUEL_TYPE_SELECTION,
	SET_FUEL_STORED_FILTERS
} from '../actionTypes'

let initialFuelTypeState = {
	isFetching: false,
	data: []
}

export const fuelType = (state = initialFuelTypeState, action) => {
	switch (action.type) {
		case REQUEST_FUEL_TYPE:
			return {
				...state,
				isFetching: true
			}

		case RECEIVE_FUEL_TYPE:
			let selectionState = state.data.reduce((acc, item) => {
				acc[item.id] = item
				return acc
			}, {})

			return {
				...state,
				isFetching: false,
				data: state.data.length ? action.data.map( item => {
					item.isSelected = (selectionState[item.id].isSelected && item.carCount > 0 )
					return item
				}) : action.data
			}

		case FUEL_TYPE_SELECTION:
			const newData = state.data.map(item => {
				if(item.id === action.id) {
					return {
						...item,
						isSelected: !item.isSelected
					}
				}
				else {
					return {
						...item
					}
				}
			})

			return {
				...state,
				data: newData
			}

			case SET_FUEL_STORED_FILTERS:
				const data = state.data.map(item => {
					if(action.storedFilters.indexOf(item.id) > -1 && item.carCount > 0) {
						return {
							...item,
							isSelected: true
						}
					}
					else {
						return {
							...item,
							isSelected: false
						}
					}
				})

				return {
					...state,
					data: data
				}

		default:
			return state
	}
}
