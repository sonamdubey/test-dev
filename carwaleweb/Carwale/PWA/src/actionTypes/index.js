// location: city
export const SELECT_CITY = "SELECT_CITY";
export const SELECT_AREA = "SELECT_AREA";
export const SET_LOCATION = "SET_LOCATION";
export const PREFILL_LOCATION = "PREFILL_LOCATION";

// toast
export const SET_TOAST = "SET_TOAST";
export const REMOVE_TOAST = "REMOVE_TOAST";

// header
export const SHOW_HEADER = "SHOW_HEADER";
export const HIDE_HEADER = "HIDE_HEADER";

// footer
export const SHOW_FOOTER = "SHOW_FOOTER";
export const HIDE_FOOTER = "HIDE_FOOTER";

// navigation drawer
export const OPEN_NAV_DRAWER = "OPEN_NAV_DRAWER";
export const CLOSE_NAV_DRAWER = "CLOSE_NAV_DRAWER";

// fullscreen
export const HIDE_HEADER_FOOTER = "HIDE_HEADER_FOOTER";
export const SHOW_HEADER_FOOTER = "SHOW_HEADER_FOOTER";

//page
export const SET_PAGE = "SET_PAGE";

// autoDetect location
export const AUTODETECT_LOCATION = "AUTODETECT_LOCATION"

export const SELECT_LATLONG = "SELECT_LATLONG";
export const FETCHING_LOCATION = "FETCHING_LOCATION"
export const SELECT_AREA_CITIES = "SELECT_AREA_CITIES"
