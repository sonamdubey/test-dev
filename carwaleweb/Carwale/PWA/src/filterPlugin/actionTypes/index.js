export const OPEN_FILTER_SCREEN = 'OPEN_FILTER_SCREEN'
export const CLOSE_FILTER_SCREEN = 'CLOSE_FILTER_SCREEN'
export const SET_FILTERS = 'SET_FILTERS'
