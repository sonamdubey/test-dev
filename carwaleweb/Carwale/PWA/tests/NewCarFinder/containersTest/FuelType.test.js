import React from 'react'
import { Redirect, MemoryRouter } from 'react-router-dom'

import FuelType from '../../../src/NewCarFinder/containers/FuelType'

import { SET_TOAST } from '../../../src/actionTypes'

import {
	SET_CURRENT_SCREEN,
	REQUEST_FUEL_TYPE,
	RECEIVE_FUEL_TYPE,
	FUEL_TYPE_SELECTION,
} from '../../../src/NewCarFinder/actionTypes'

const fuelTypeApiResponse = [
    {"id":1,"name":"Petrol","carCount":12,"description":"Indian auto industry offers several car models powered by engines running on petrol, diesel, CNG and LPG.","icon":"https://imgd.aeplcdn.com/0x0/cw/fuel/svg/petrol.svg","isSelected":false},
    {"id":2,"name":"Diesel","carCount":13,"description":"Get the list of cars Under 3 lakhs and their on road price, mileage, rating review. C\nheck out the list of best diesel, compact suv, sedan, hatchback cars below 3 lakhs available in India.","icon":"https://imgd.aeplcdn.com/0x0/cw/fuel/svg/diesel.svg","isSelected":false},
    {"id":3,"name":"CNG","carCount":0,"description":null,"icon":"https://imgd.aeplcdn.com/0x0/cw/fuel/svg/cng.svg","isSelected":false},
    {"id":4,"name":"LPG","carCount":0,"description":null,"icon":"https://imgd.aeplcdn.com/0x0/cw/fuel/svg/lpg.svg","isSelected":false},
    {"id":5,"name":"Electric","carCount":0,"description":null,"icon":"https://imgd.aeplcdn.com/0x0/cw/fuel/svg/electric.svg","isSelected":false}
]


// Mock body type api
jest.mock('../../../src/NewCarFinder/apis/FuelType', () => {
    const fuelTypeApiResponse = [
        {"id":1,"name":"Petrol","carCount":12,"description":"Indian auto industry offers several car models powered by engines running on petrol, diesel, CNG and LPG.","icon":"https://imgd.aeplcdn.com/0x0/cw/fuel/svg/petrol.svg","isSelected":false},
        {"id":2,"name":"Diesel","carCount":13,"description":"Get the list of cars Under 3 lakhs and their on road price, mileage, rating review. C\nheck out the list of best diesel, compact suv, sedan, hatchback cars below 3 lakhs available in India.","icon":"https://imgd.aeplcdn.com/0x0/cw/fuel/svg/diesel.svg","isSelected":false},
        {"id":3,"name":"CNG","carCount":0,"description":null,"icon":"https://imgd.aeplcdn.com/0x0/cw/fuel/svg/cng.svg","isSelected":false},
        {"id":4,"name":"LPG","carCount":0,"description":null,"icon":"https://imgd.aeplcdn.com/0x0/cw/fuel/svg/lpg.svg","isSelected":false},
        {"id":5,"name":"Electric","carCount":0,"description":null,"icon":"https://imgd.aeplcdn.com/0x0/cw/fuel/svg/electric.svg","isSelected":false}
    ]
    return {
        get: jest.fn((options) => Promise.resolve(fuelTypeApiResponse))
    }
})

//Mock Ripple js
jest.mock('../../../src/utils/Ripple')

describe("Tests for FuelType container component", () => {

    let props
    let store
    let enzymeWrapper
    let state
    const Wrapper = () => {
        if (!enzymeWrapper) {
            store = mockStore(state)
            enzymeWrapper = shallowWithStore(<FuelType {...props} />, store).dive()
        }
        return enzymeWrapper
    }

    beforeEach(() => {
        props = undefined
        state = getMockInitialStore()
        enzymeWrapper = undefined
        store = undefined
    })

    describe("when no fuel type data is available", () => {
        it("should match snapshot", () => {
            const wrapper = Wrapper()
            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("when component is mounted", () => {
        it("should dispatch appropriate actions", (done) => {
            /**
             * documentation for done and use case : https://jasmine.github.io/2.5/introduction.html#section-58
             * and https://stackoverflow.com/questions/47275706/async-function-not-behaving-as-i-expect-with-jest
             */
            const wrapper = Wrapper()
            const expectedActions = store.getActions()
            expect(expectedActions).toHaveLength(2)

            expect(expectedActions).toContainEqual({
                type: SET_CURRENT_SCREEN,
                screenId: 2
            })

            expect(expectedActions).toContainEqual({
                type: REQUEST_FUEL_TYPE
            })

            // http://www.bambielli.com/til/2017-09-19-process-nexttick/
            // Please read this blog before attempting a change : http://voidcanvas.com/setimmediate-vs-nexttick-vs-settimeout/
            process.nextTick(() => {
                setTimeout(() => {
                    try {
                        expect(store.getActions()).toHaveLength(3)
                        expect(store.getActions()).toContainEqual({
                            type: RECEIVE_FUEL_TYPE,
                            data: fuelTypeApiResponse
                        })
                    } catch (e) {
                        return done(e);
                    }
                    done();
                },300)
            });
        })
    })

    describe("when data is available", () => {
        it("should match snapshot", () => {
            state.newCarFinder.fuelType.data = fuelTypeApiResponse
            const wrapper = Wrapper()
            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("when fuel type with count > 0 is clicked", () => {
        it("should dispatch FUEL_TYPE_SELECTION action", () => {
            state.newCarFinder.fuelType.data = fuelTypeApiResponse
            const wrapper = Wrapper()
            let activeItems = fuelTypeApiResponse.filter(x => x.carCount > 0)
            if (activeItems.length) {
                const firstItem = wrapper.find('.fuel-type-list__item').findWhere(x => x.key() == activeItems[0].id)
                firstItem.simulate('click')
                const expectedActions = store.getActions()
                expect(expectedActions).toContainEqual({
                    type: FUEL_TYPE_SELECTION,
                    id: parseInt(firstItem.key())
                })
            }
        })
    })

    describe("when fuel type with count == 0 is clicked", () => {
        it("should not dispatch FUEL_TYPE_SELECTION action", () => {
            state.newCarFinder.fuelType.data = fuelTypeApiResponse
            const wrapper = Wrapper()
            let disabledItems = fuelTypeApiResponse.filter(x => x.carCount == 0)
            if (disabledItems.length) {
                const firstItem = wrapper.find('.fuel-type-list__item').findWhere(x => x.key() == disabledItems[0].id)
                firstItem.simulate('click')
                const expectedActions = store.getActions().filter(x => x.type == SET_TOAST)
                expect(expectedActions.length).toBe(1)
                //expect(expectedActions[0].message).toBe(``)
            }
        })
    })

})
