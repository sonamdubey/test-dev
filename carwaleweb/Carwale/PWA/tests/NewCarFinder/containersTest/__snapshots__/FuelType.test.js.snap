// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for FuelType container component when data is available should match snapshot 1`] = `
<div
  className="fuel-type-screen"
>
  <div
    className="screen__head"
  >
    <p
      className="screen-head__title"
    >
      Select your preferred fuel type
    </p>
  </div>
  <div
    className="screen__body"
  >
    <ul
      className="fuel-type__list"
    >
      <li
        className="fuel-type-list__item "
        key="1"
        onClick={[Function]}
      >
        <ListItem
          item={
            Object {
              "carCount": 12,
              "description": "Indian auto industry offers several car models powered by engines running on petrol, diesel, CNG and LPG.",
              "icon": "https://imgd.aeplcdn.com/0x0/cw/fuel/svg/petrol.svg",
              "id": 1,
              "isSelected": false,
              "name": "Petrol",
            }
          }
          showDescription={true}
        />
      </li>
      <li
        className="fuel-type-list__item "
        key="2"
        onClick={[Function]}
      >
        <ListItem
          item={
            Object {
              "carCount": 13,
              "description": "Get the list of cars Under 3 lakhs and their on road price, mileage, rating review. C
heck out the list of best diesel, compact suv, sedan, hatchback cars below 3 lakhs available in India.",
              "icon": "https://imgd.aeplcdn.com/0x0/cw/fuel/svg/diesel.svg",
              "id": 2,
              "isSelected": false,
              "name": "Diesel",
            }
          }
          showDescription={true}
        />
      </li>
      <li
        className="fuel-type-list__item disable"
        key="3"
        onClick={[Function]}
      >
        <ListItem
          item={
            Object {
              "carCount": 0,
              "description": null,
              "icon": "https://imgd.aeplcdn.com/0x0/cw/fuel/svg/cng.svg",
              "id": 3,
              "isSelected": false,
              "name": "CNG",
            }
          }
          showDescription={true}
        />
      </li>
      <li
        className="fuel-type-list__item disable"
        key="4"
        onClick={[Function]}
      >
        <ListItem
          item={
            Object {
              "carCount": 0,
              "description": null,
              "icon": "https://imgd.aeplcdn.com/0x0/cw/fuel/svg/lpg.svg",
              "id": 4,
              "isSelected": false,
              "name": "LPG",
            }
          }
          showDescription={true}
        />
      </li>
      <li
        className="fuel-type-list__item disable"
        key="5"
        onClick={[Function]}
      >
        <ListItem
          item={
            Object {
              "carCount": 0,
              "description": null,
              "icon": "https://imgd.aeplcdn.com/0x0/cw/fuel/svg/electric.svg",
              "id": 5,
              "isSelected": false,
              "name": "Electric",
            }
          }
          showDescription={true}
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`Tests for FuelType container component when no fuel type data is available should match snapshot 1`] = `
<div
  className="fuel-type-screen"
>
  <div
    className="screen__head"
  >
    <p
      className="screen-head__title"
    >
      Select your preferred fuel type
    </p>
  </div>
  <div
    className="screen__body"
  >
    <ul
      className="fuel-type__list"
    />
  </div>
</div>
`;
